types/src/validator_verifier.rs

```diff
@@ -79,77 +84,96 @@ impl ValidatorConsensusInfo {
 /// Supports validation of signatures for known authors with individual voting powers. This struct
 /// can be used for all signature verification operations including block and network signature
 /// verification, respectively.
-#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
-#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
+#[derive(Clone, Debug, Eq, PartialEq, Serialize)]
 pub struct ValidatorVerifier {
-    /// An ordered map of each validator's on-chain account address to its pubkeys
-    /// and voting power.
-    address_to_validator_info: BTreeMap<AccountAddress, ValidatorConsensusInfo>,
+    /// A vector of each validator's on-chain account address to its pubkeys and voting power.
+    validator_infos: Vec<ValidatorConsensusInfo>,
```

Comment:
should we have index in ValidatorConsensusInfo, so that this class can assert in constructor that order in validator_infos is corect?

Reply:
but the Arbitrary is hard to fix, it generates arbitrary data for each field

Reply:
instead I did the assertion on the move -> rust conversion